app/src/main/java/org/xbmc/kore/ui/generic/SubtitlesOptionsDialog.java

```diff
@@ -0,0 +1,313 @@
+package org.xbmc.kore.ui.generic;
+
+
+import android.app.Activity;
+
+import android.os.Handler;
+import android.support.v4.app.FragmentActivity;
+import android.text.TextUtils;
+import android.widget.Toast;
+
+import org.xbmc.kore.R;
+import org.xbmc.kore.host.HostConnectionObserver;
+import org.xbmc.kore.host.HostInfo;
+import org.xbmc.kore.host.HostManager;
+import org.xbmc.kore.jsonrpc.ApiCallback;
+import org.xbmc.kore.jsonrpc.ApiMethod;
+import org.xbmc.kore.jsonrpc.HostConnection;
+import org.xbmc.kore.jsonrpc.method.Addons;
+import org.xbmc.kore.jsonrpc.method.GUI;
+import org.xbmc.kore.jsonrpc.method.Input;
+import org.xbmc.kore.jsonrpc.method.Player;
+import org.xbmc.kore.jsonrpc.type.ListType;
+import org.xbmc.kore.jsonrpc.type.PlayerType;
+import org.xbmc.kore.utils.LogUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * Dialog that presents a list of options related to subtitles.
+ */
+public class SubtitlesOptionsDialog
+        implements HostConnectionObserver.PlayerEventsObserver,
```

Comment:
The reason the dialog is listening the player’s events is to be able to have an up to date list of the available subtitles. I think a better solution to this is to create a function updateSubtitleList that will be called by the activity in order to provide the dialog object with the latest list of subtitles when needed.